src/server/conn/auto.rs

```diff
@@ -52,17 +81,23 @@ impl<E> Builder<E> {
     /// ```
     pub fn new(executor: E) -> Self {
         Self {
+            #[cfg(feature = "http1")]
             http1: http1::Builder::new(),
+            #[cfg(feature = "http2")]
             http2: http2::Builder::new(executor),
+            #[cfg(not(feature = "http2"))]
+            _executor: executor,
         }
     }
 
     /// Http1 configuration.
+    #[cfg(feature = "http1")]
```

Comment:
I think the main blocker is that this would be breaking, no? Currently, if you forgot to enable the `http1` or `http2` features, after this change, your code would no longer compile.

Reply:
The only way to have this be publicly exposed before is by enabling `server-auto`. That feature now enables `http1` and `http2`, making this backwards compatible.

Reply:
Oh right, sorry. I just realized reading back I understood that in my previous comment.

OK, so with this PR, if you only enable `server` and `http1`, the auto builder can exist, with only the `Http1Builder` part of it. Does that sound right?

Reply:
That's right.